File: backend/gym-service/src/main/java/com/fitconnect/gym/entity/Gym.java
package com.fitconnect.gym.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "gym")
@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
public class Gym {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    // Reference only — the actual user record lives in auth-service's DB.
    // No JPA relationship across services; this is just an opaque foreign id from the JWT.
    @Column(nullable = false)
    private Long ownerId;

    @Column(nullable = false)
    private String name;

    @Column(length = 1000)
    private String description;

    @Column(nullable = false)
    private String address;

    private Double latitude;
    private Double longitude;

    private LocalTime openingTime;
    private LocalTime closingTime;

    private String contactPhone;
    private String contactEmail;

    private Integer establishedYear;

    @Builder.Default
    @OneToMany(mappedBy = "gym", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<GymMedia> media = new ArrayList<>();

    @Builder.Default
    @OneToMany(mappedBy = "gym", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<FeePlan> feePlans = new ArrayList<>();

    @Builder.Default
    @OneToMany(mappedBy = "gym", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Achievement> achievements = new ArrayList<>();
}

File: frontend/src/types/gym.ts
export interface FeePlan {
  id: number;
  planName: string;
  amount: number;
  billingCycle: string;
}

export interface GymMedia {
  id: number;
  url: string;
  type: string;
  isCover: boolean;
}

export interface Achievement {
  id: number;
  title: string;
  description: string;
  achievedOn: string;
}

export interface GymPublicResponse {
  id: number;
  name: string;
  description: string;
  address: string;
  latitude: number | null;
  longitude: number | null;
  openingTime: string | null;
  closingTime: string | null;
  contactPhone: string;
  contactEmail: string;
  establishedYear: number | null;
  media: GymMedia[];
  feePlans: FeePlan[];
  achievements: Achievement[];
}